include/threads/ThreadState.h

```diff
@@ -57,11 +56,44 @@ class Level
     std::condition_variable nowaitCv;
 };
 
+class PthreadTask
+{
+  public:
+    PthreadTask(faabric::Message* parentMsgIn,
+                std::shared_ptr<faabric::Message> msgIn)
+      : parentMsg(parentMsgIn)
+      , msg(msgIn)
+    {}
+
+    bool isShutdown = false;
+    faabric::Message* parentMsg;
```

Comment:
Unfortunately the parent message comes from `getExecutingCall` which is used all over the place and returns a raw pointer. We know that the lifetime of this message exceeds the threaded call though, so it's ok for now.